<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
          "DTD/xhtml1-strict.dtd">
<html>
  

  <head>
    
    <title>twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests</title>
    <meta name="generator" content="pydoctor 22.2.2.dev0"> 
        
    </meta>
    <meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=0.75" />
    <link rel="stylesheet" type="text/css" href="bootstrap.min.css" />
    <link rel="stylesheet" type="text/css" href="apidocs.css" />
    <link rel="stylesheet" type="text/css" href="extra.css" />
</head>

  <body>

    <div id="banner">    
    <div>
        <a href="/">Home</a>
        &gt; twisted-22.1.0 <!-- This is a placeholder -->
        
        <!-- (<a href=""&gt;show all versions</a&gt;) -->
    </div>
</div>

    <nav class="navbar navbar-default">
  
  <div class="container">

    <div class="navbar-header">
      
      <div class="navlinks">
        <span class="navbar-brand">
          twisted <a href="index.html">API Documentation</a>
        </span>

        <a href="moduleIndex.html">
          Modules
        </a>

        <a href="classIndex.html">
          Classes
        </a>

        <a href="nameIndex.html">
          Names
        </a>

        <div id="search-box-container">
          <div class="input-group">
            <input id="search-box" type="search" name="search-query" placeholder="Search..." aria-label="Search" minlength="2" class="form-control" autocomplete="off" />
            
            <span class="input-group-btn">
              <a style="display: none;" class="btn btn-default" id="search-clear-button" title="Clear" onclick="clearSearch()"><img src="fonts/x-circle.svg" alt="Clear" /></a>
              <a class="btn btn-default" id="search-help-button" title="Help" onclick="toggleSearchHelpText()"><img src="fonts/info.svg" alt="Help" /></a>
            </span>
          </div>
        </div>

      </div>

      <div id="search-results-container" style="display: none;">
        <noscript>
            <h1>Cannot search: JavaScript is not supported/enabled in your browser.</h1>
        </noscript>

        <div class="hint" id="search-help-box">
          <p class="rst-last">
      
            Search bar offers the following options:
            <ul>   
                <li>
                  <strong>Term presence.</strong> The below example searches for documents that 
                    must contain “foo”, might contain “bar” and must not contain “baz”: <code>+foo bar -baz</code>
                </li> 

                <li>
                  <strong>Wildcards.</strong> The below example searches for documents with words beginning with “foo”: <code>foo*</code>
                </li> 

                <li>
                  <strong>Search in specific fields.</strong> The following search matches all documents 
                  that contains "foo" in their full dotted name and with “bar” in any field: <code>+fullName:*foo* +bar</code>

                  <p>
                    Possible fields: 'name', 'docstring', 'fullName'.
                  </p>
                </li>

                <li>
                  <strong>Fuzzy matches.</strong> The following search matches all documents 
                  that have a word within 1 edit distance of “foo”: <code>foo~1</code>
                </li>
            </ul>

          </p>
        </div>

        <div id="search-status"> </div>
        
        <div class="warning" id="search-warn-box" style="display: none;">
          <p class="rst-last"><span id="search-warn"></span></p>
        </div>

        <table id="search-results">
          <!-- Filled dynamically by JS -->
        </table>
        
        <div>
          <br />
          <p>Results provided by <a href="https://lunrjs.com">Lunr.js</a></p>
        </div>
      </div>

    </div>
  </div>

  <script src="ajax.js" type="text/javascript"></script>
  <script src="search.js" type="text/javascript"></script>

</nav>

    

    <div class="container">

      <div class="page-header">
        <h1 class="class"><code><code><a href="index.html" class="internal-link">twisted</a></code><wbr></wbr>.<code><a href="twisted.trial.html" class="internal-link" title="twisted.trial">trial</a></code><wbr></wbr>.<code><a href="twisted.trial.test.html" class="internal-link" title="twisted.trial.test">test</a></code><wbr></wbr>.<code><a href="twisted.trial.test.test_assertions.html" class="internal-link" title="twisted.trial.test.test_assertions">test_assertions</a></code><wbr></wbr>.<code><a href="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.html" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests">ResultOfCoroutineAssertionsTests</a></code></code></h1>
        <div id="showPrivate">
          <button class="btn btn-link" onclick="togglePrivate()">Toggle Private API</button>
        </div>
      </div>

      <div class="categoryHeader">
        class documentation
      </div>

      <div class="extrasDocstring">
        <p><code><span class="py-keyword">class</span> <span class="py-defname">ResultOfCoroutineAssertionsTests</span>(<a href="twisted.trial.unittest.SynchronousTestCase.html" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase">unittest.SynchronousTestCase</a>): <a href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1188" class="sourceLink">(source)</a></code></p>
        <p><a href="classIndex.html#twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests">View In Hierarchy</a></p>
      </div>

      <div class="moduleDocstring">
        <div>Tests for <code><a href="twisted.trial._synctest._Assertions.html#successResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.successResultOf">SynchronousTestCase.successResultOf</a></code>, <code><a href="twisted.trial._synctest._Assertions.html#failureResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.failureResultOf">SynchronousTestCase.failureResultOf</a></code>, and <code><a href="twisted.trial._synctest._Assertions.html#assertNoResult" class="internal-link" title="twisted.trial._synctest._Assertions.assertNoResult">SynchronousTestCase.assertNoResult</a></code> when given a coroutine.</div>
      </div>

      <div id="splitTables">
        <table class="children sortable" id="id10906">
  
  
  <tr class="method">
    
    <td>Async Method</td>
    <td><code><a href="#noCurrentResult" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.noCurrentResult">no​Current​Result</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="method">
    
    <td>Async Method</td>
    <td><code><a href="#raisesException" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.raisesException">raises​Exception</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="method">
    
    <td>Async Method</td>
    <td><code><a href="#successResult" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.successResult">success​Result</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#test_assertNoResult" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_assertNoResult">test_assert​No​Result</a></code></td>
    <td>When passed a coroutine with no current result, <code><a href="twisted.trial._synctest._Assertions.html#assertNoResult" class="internal-link" title="twisted.trial._synctest._Assertions.assertNoResult">SynchronousTestCase.assertNoResult</a></code> does not raise an exception.</td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#test_assertNoResultFailure" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_assertNoResultFailure">test_assert​No​Result​Failure</a></code></td>
    <td>When passed a coroutine which currently has an exception result (see <code>test_withFailureResult</code>), <code><a href="twisted.trial._synctest._Assertions.html#assertNoResult" class="internal-link" title="twisted.trial._synctest._Assertions.assertNoResult">SynchronousTestCase.assertNoResult</a></code> raises <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code>.</td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#test_assertNoResultPropagatesLaterFailure" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_assertNoResultPropagatesLaterFailure">test_assert​No​Result​Propagates​Later​Failure</a></code></td>
    <td><span class="undocumented">No summary</span></td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#test_assertNoResultPropagatesSuccess" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_assertNoResultPropagatesSuccess">test_assert​No​Result​Propagates​Success</a></code></td>
    <td><span class="undocumented">No summary</span></td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#test_assertNoResultSuccess" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_assertNoResultSuccess">test_assert​No​Result​Success</a></code></td>
    <td>When passed a coroutine which currently has a success result (see <code>test_withSuccessResult</code>), <code><a href="twisted.trial._synctest._Assertions.html#assertNoResult" class="internal-link" title="twisted.trial._synctest._Assertions.assertNoResult">SynchronousTestCase.assertNoResult</a></code> raises <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code>.</td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#test_assertNoResultSwallowsImmediateFailure" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_assertNoResultSwallowsImmediateFailure">test_assert​No​Result​Swallows​Immediate​Failure</a></code></td>
    <td>When passed a <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.internet.defer.Deferred.html" class="intersphinx-link">Deferred</a></code> which currently has a <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.python.failure.Failure.html" class="intersphinx-link">Failure</a></code> result, <code><a href="twisted.trial._synctest._Assertions.html#assertNoResult" class="internal-link" title="twisted.trial._synctest._Assertions.assertNoResult">SynchronousTestCase.assertNoResult</a></code> changes the result of the <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.internet.defer.Deferred.html" class="intersphinx-link">Deferred</a></code> to a success.</td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#test_failureResultOfWithException" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_failureResultOfWithException">test_failure​Result​Of​With​Exception</a></code></td>
    <td><span class="undocumented">No summary</span></td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#test_failureResultOfWithExpectedException" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_failureResultOfWithExpectedException">test_failure​Result​Of​With​Expected​Exception</a></code></td>
    <td><span class="undocumented">No summary</span></td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#test_failureResultOfWithoutResult" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_failureResultOfWithoutResult">test_failure​Result​Of​Without​Result</a></code></td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#failureResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.failureResultOf">SynchronousTestCase.failureResultOf</a></code> raises <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code> when called with a coroutine with no current result.</td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#test_failureResultOfWithSuccess" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_failureResultOfWithSuccess">test_failure​Result​Of​With​Success</a></code></td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#failureResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.failureResultOf">SynchronousTestCase.failureResultOf</a></code> raises <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code> when called with a coroutine with a success result.</td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#test_failureResultOfWithWrongExceptionMultiExpectedExceptions" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_failureResultOfWithWrongExceptionMultiExpectedExceptions">test_failure​Result​Of​With​Wrong​Exception​Multi​Expected​Exceptions</a></code></td>
    <td><span class="undocumented">No summary</span></td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#test_failureResultOfWithWrongExceptionMultiExpectedExceptionsHasTB" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_failureResultOfWithWrongExceptionMultiExpectedExceptionsHasTB">test_failure​Result​Of​With​Wrong​Exception​Multi​Expected​Exceptions​Has​TB</a></code></td>
    <td><span class="undocumented">No summary</span></td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#test_failureResultOfWithWrongExceptionOneExpectedException" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_failureResultOfWithWrongExceptionOneExpectedException">test_failure​Result​Of​With​Wrong​Exception​One​Expected​Exception</a></code></td>
    <td><span class="undocumented">No summary</span></td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#test_failureResultOfWithWrongExceptionOneExpectedExceptionHasTB" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_failureResultOfWithWrongExceptionOneExpectedExceptionHasTB">test_failure​Result​Of​With​Wrong​Exception​One​Expected​Exception​Has​TB</a></code></td>
    <td><span class="undocumented">No summary</span></td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#test_failureResultOfWithWrongFailure" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_failureResultOfWithWrongFailure">test_failure​Result​Of​With​Wrong​Failure</a></code></td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#failureResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.failureResultOf">SynchronousTestCase.failureResultOf</a></code> raises <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code> when called with a coroutine that raises an exception that was not expected.</td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#test_successResultOfWithException" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_successResultOfWithException">test_success​Result​Of​With​Exception</a></code></td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#successResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.successResultOf">SynchronousTestCase.successResultOf</a></code> raises <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code> when called with a coroutine that raises an exception.</td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#test_successResultOfWithFailureHasTraceback" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_successResultOfWithFailureHasTraceback">test_success​Result​Of​With​Failure​Has​Traceback</a></code></td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#successResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.successResultOf">SynchronousTestCase.successResultOf</a></code> raises a <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code> that has the original failure traceback when called with a coroutine with a failure result.</td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#test_successResultOfWithSuccessResult" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_successResultOfWithSuccessResult">test_success​Result​Of​With​Success​Result</a></code></td>
    <td><span class="undocumented">No summary</span></td>
  </tr><tr class="method">
    
    <td>Method</td>
    <td><code><a href="#test_withoutResult" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_withoutResult">test_without​Result</a></code></td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#successResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.successResultOf">SynchronousTestCase.successResultOf</a></code> raises <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code> when called with a coroutine with no current result.</td>
  </tr><tr class="classvariable">
    
    <td>Class Variable</td>
    <td><code><a href="#exception" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.exception">exception</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="classvariable">
    
    <td>Class Variable</td>
    <td><code><a href="#failure" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.failure">failure</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="classvariable">
    
    <td>Class Variable</td>
    <td><code><a href="#result" class="internal-link" title="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.result">result</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr>
</table>
        
          <p class="inheritedFrom">
            Inherited from <code><a href="twisted.trial.unittest.SynchronousTestCase.html" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase">SynchronousTestCase</a></code>:
          </p>
          <table class="children sortable" id="id10907">
  
  
  <tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#__eq__" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.__eq__">__eq__</a></code></td>
    <td><span class="undocumented">No summary</span></td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#__hash__" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.__hash__">__hash__</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#__init__" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.__init__">__init__</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#addCleanup" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.addCleanup">add​Cleanup</a></code></td>
    <td>Add the given function to a list of functions to be called after the test has run, but before <tt class="rst-docutils literal">tearDown</tt>.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#callDeprecated" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.callDeprecated">call​Deprecated</a></code></td>
    <td>Call a function that should have been deprecated at a specific version and in favor of a specific alternative, and assert that it was thusly deprecated.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#flushLoggedErrors" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.flushLoggedErrors">flush​Logged​Errors</a></code></td>
    <td>Remove stored errors received from the log.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#flushWarnings" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.flushWarnings">flush​Warnings</a></code></td>
    <td>Remove stored warnings from the list of captured warnings and return them.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#getDeprecatedModuleAttribute" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.getDeprecatedModuleAttribute">get​Deprecated​Module​Attribute</a></code></td>
    <td>Retrieve a module attribute which should have been deprecated, and assert that we saw the appropriate deprecation warning.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#getSkip" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.getSkip">get​Skip</a></code></td>
    <td><span class="undocumented">No summary</span></td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#getTodo" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.getTodo">get​Todo</a></code></td>
    <td><span class="undocumented">No summary</span></td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#mktemp" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.mktemp">mktemp</a></code></td>
    <td>Create a new path name which can be used for a new file or directory.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#patch" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.patch">patch</a></code></td>
    <td>Monkey patch an object for the duration of the test.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#run" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.run">run</a></code></td>
    <td>Run the test case, storing the results in <tt class="rst-docutils literal">result</tt>.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#runTest" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.runTest">run​Test</a></code></td>
    <td>If no <tt class="rst-docutils literal">methodName</tt> argument is passed to the constructor, <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.trial.unittest.SynchronousTestCase.html#run" class="intersphinx-link">run</a></code> will treat this method as the thing with the actual test inside.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#shortDescription" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.shortDescription">short​Description</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="baseinstancevariable">
    
    <td>Instance Variable</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">failure​Exception</a></code></td>
    <td>An exception class, defaulting to <tt class="rst-docutils literal">FailTest</tt>. If the test method raises this exception, it will be reported as a failure, rather than an exception. All of the assertion methods raise this if the assertion fails.</td>
  </tr><tr class="baseinstancevariable">
    
    <td>Instance Variable</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#skip" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.skip">skip</a></code></td>
    <td><code><a href="https://docs.python.org/3/library/constants.html#None" class="intersphinx-link">None</a></code> or a string explaining why this test is to be skipped. If defined, the test will not be run. Instead, it will be reported to the result object as 'skipped' (if the <tt class="rst-docutils literal">TestResult</tt> supports skipping).</td>
  </tr><tr class="baseinstancevariable">
    
    <td>Instance Variable</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#suppress" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.suppress">suppress</a></code></td>
    <td><code><a href="https://docs.python.org/3/library/constants.html#None" class="intersphinx-link">None</a></code> or a list of tuples of <tt class="rst-docutils literal">(args, kwargs)</tt> to be passed to <tt class="rst-docutils literal">warnings.filterwarnings</tt>. Use these to suppress warnings raised in a test. Useful for testing deprecated code. See also <code><a href="twisted.trial.util.html#suppress" class="internal-link" title="twisted.trial.util.suppress">util.suppress</a></code>.</td>
  </tr><tr class="baseinstancevariable">
    
    <td>Instance Variable</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#todo" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.todo">todo</a></code></td>
    <td><code><a href="https://docs.python.org/3/library/constants.html#None" class="intersphinx-link">None</a></code>, a string or a tuple of <tt class="rst-docutils literal">(errors, reason)</tt> where <tt class="rst-docutils literal">errors</tt> is either an exception class or an iterable of exception classes, and <tt class="rst-docutils literal">reason</tt> is a string. See <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.trial.unittest.Todo.html" class="intersphinx-link">Todo</a></code> or <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.trial.unittest.html#makeTodo" class="intersphinx-link">makeTodo</a></code> for more information.</td>
  </tr><tr class="basemethod private">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#_getSkipReason" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase._getSkipReason">_get​Skip​Reason</a></code></td>
    <td>Return the reason to use for skipping a test method.</td>
  </tr><tr class="basemethod private">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#_getSuppress" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase._getSuppress">_get​Suppress</a></code></td>
    <td><span class="undocumented">No summary</span></td>
  </tr><tr class="basemethod private">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#_installObserver" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase._installObserver">_install​Observer</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="basemethod private">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#_removeObserver" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase._removeObserver">_remove​Observer</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="basemethod private">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#_run" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase._run">​_run</a></code></td>
    <td>Run a single method, either a test method or fixture.</td>
  </tr><tr class="basemethod private">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#_runCleanups" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase._runCleanups">_run​Cleanups</a></code></td>
    <td>Synchronously run any cleanups which have been added.</td>
  </tr><tr class="basemethod private">
    
    <td>Method</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#_runFixturesAndTest" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase._runFixturesAndTest">_run​Fixtures​And​Test</a></code></td>
    <td>Run <tt class="rst-docutils literal">setUp</tt>, a test method, test cleanups, and <tt class="rst-docutils literal">tearDown</tt>.</td>
  </tr><tr class="baseinstancevariable private">
    
    <td>Instance Variable</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#_cleanups" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase._cleanups">​_cleanups</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="baseinstancevariable private">
    
    <td>Instance Variable</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#_observer" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase._observer">​_observer</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="baseinstancevariable private">
    
    <td>Instance Variable</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#_parents" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase._parents">​_parents</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="baseinstancevariable private">
    
    <td>Instance Variable</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#_passed" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase._passed">​_passed</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="baseinstancevariable private">
    
    <td>Instance Variable</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#_testMethodName" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase._testMethodName">_test​Method​Name</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="baseinstancevariable private">
    
    <td>Instance Variable</td>
    <td><code><a href="twisted.trial.unittest.SynchronousTestCase.html#_warnings" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase._warnings">​_warnings</a></code></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr>
</table>
          
          <p class="inheritedFrom">
            Inherited from <code><a href="twisted.trial._synctest._Assertions.html" class="internal-link" title="twisted.trial._synctest._Assertions">_Assertions</a></code> (via <code><a href="twisted.trial.unittest.SynchronousTestCase.html" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase">SynchronousTestCase</a></code>):
          </p>
          <table class="children sortable" id="id10908">
  
  
  <tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#assertAlmostEqual" class="internal-link" title="twisted.trial._synctest._Assertions.assertAlmostEqual">assert​Almost​Equal</a></code></td>
    <td>Fail if the two objects are unequal as determined by their difference rounded to the given number of decimal places (default 7) and comparing to zero.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#assertApproximates" class="internal-link" title="twisted.trial._synctest._Assertions.assertApproximates">assert​Approximates</a></code></td>
    <td>Fail if <tt class="rst-docutils literal">first</tt> - <tt class="rst-docutils literal">second</tt> &gt; <tt class="rst-docutils literal">tolerance</tt></td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#assertEqual" class="internal-link" title="twisted.trial._synctest._Assertions.assertEqual">assert​Equal</a></code></td>
    <td>Fail the test if <tt class="rst-docutils literal">first</tt> and <tt class="rst-docutils literal">second</tt> are not equal.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#assertFalse" class="internal-link" title="twisted.trial._synctest._Assertions.assertFalse">assert​False</a></code></td>
    <td>Fail the test if <tt class="rst-docutils literal">condition</tt> evaluates to True.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#assertIn" class="internal-link" title="twisted.trial._synctest._Assertions.assertIn">assert​In</a></code></td>
    <td>Fail the test if <tt class="rst-docutils literal">containee</tt> is not found in <tt class="rst-docutils literal">container</tt>.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#assertIs" class="internal-link" title="twisted.trial._synctest._Assertions.assertIs">assert​Is</a></code></td>
    <td>Fail the test if <tt class="rst-docutils literal">first</tt> is not <tt class="rst-docutils literal">second</tt>.  This is an obect-identity-equality test, not an object equality (i.e. <tt class="rst-docutils literal">__eq__</tt>) test.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#assertIsInstance" class="internal-link" title="twisted.trial._synctest._Assertions.assertIsInstance">assert​Is​Instance</a></code></td>
    <td>Fail if <tt class="rst-docutils literal">instance</tt> is not an instance of the given class or of one of the given classes.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#assertIsNot" class="internal-link" title="twisted.trial._synctest._Assertions.assertIsNot">assert​Is​Not</a></code></td>
    <td>Fail the test if <tt class="rst-docutils literal">first</tt> is <tt class="rst-docutils literal">second</tt>.  This is an obect-identity-equality test, not an object equality (i.e. <tt class="rst-docutils literal">__eq__</tt>) test.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#assertNoResult" class="internal-link" title="twisted.trial._synctest._Assertions.assertNoResult">assert​No​Result</a></code></td>
    <td>Assert that <tt class="rst-docutils literal">deferred</tt> does not have a result at this point.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#assertNotAlmostEqual" class="internal-link" title="twisted.trial._synctest._Assertions.assertNotAlmostEqual">assert​Not​Almost​Equal</a></code></td>
    <td>Fail if the two objects are equal as determined by their difference rounded to the given number of decimal places (default 7) and comparing to zero.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#assertNotEqual" class="internal-link" title="twisted.trial._synctest._Assertions.assertNotEqual">assert​Not​Equal</a></code></td>
    <td>Fail the test if <tt class="rst-docutils literal">first</tt> == <tt class="rst-docutils literal">second</tt>.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#assertNotIn" class="internal-link" title="twisted.trial._synctest._Assertions.assertNotIn">assert​Not​In</a></code></td>
    <td>Fail the test if <tt class="rst-docutils literal">containee</tt> is found in <tt class="rst-docutils literal">container</tt>.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#assertNotIsInstance" class="internal-link" title="twisted.trial._synctest._Assertions.assertNotIsInstance">assert​Not​Is​Instance</a></code></td>
    <td>Fail if <tt class="rst-docutils literal">instance</tt> is an instance of the given class or of one of the given classes.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#assertNotSubstring" class="internal-link" title="twisted.trial._synctest._Assertions.assertNotSubstring">assert​Not​Substring</a></code></td>
    <td>Fail if <tt class="rst-docutils literal">astring</tt> contains <tt class="rst-docutils literal">substring</tt>.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#assertRaises" class="internal-link" title="twisted.trial._synctest._Assertions.assertRaises">assert​Raises</a></code></td>
    <td>Fail the test unless calling the function <tt class="rst-docutils literal">f</tt> with the given <tt class="rst-docutils literal">args</tt> and <tt class="rst-docutils literal">kwargs</tt> raises <tt class="rst-docutils literal">exception</tt>. The failure will report the traceback and call stack of the unexpected exception.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#assertSubstring" class="internal-link" title="twisted.trial._synctest._Assertions.assertSubstring">assert​Substring</a></code></td>
    <td>Fail if <tt class="rst-docutils literal">substring</tt> does not exist within <tt class="rst-docutils literal">astring</tt>.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#assertTrue" class="internal-link" title="twisted.trial._synctest._Assertions.assertTrue">assert​True</a></code></td>
    <td>Fail the test if <tt class="rst-docutils literal">condition</tt> evaluates to False.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#assertWarns" class="internal-link" title="twisted.trial._synctest._Assertions.assertWarns">assert​Warns</a></code></td>
    <td>Fail if the given function doesn't generate the specified warning when called. It calls the function, checks the warning, and forwards the result of the function if everything is fine.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#fail" class="internal-link" title="twisted.trial._synctest._Assertions.fail">fail</a></code></td>
    <td>Absolutely fail the test.  Do not pass go, do not collect $200.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#failureResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.failureResultOf">failure​Result​Of</a></code></td>
    <td>Return the current failure result of <tt class="rst-docutils literal">deferred</tt> or raise <tt class="rst-docutils literal">self.failureException</tt>.</td>
  </tr><tr class="basemethod">
    
    <td>Method</td>
    <td><code><a href="twisted.trial._synctest._Assertions.html#successResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.successResultOf">success​Result​Of</a></code></td>
    <td>Return the current success result of <tt class="rst-docutils literal">deferred</tt> or raise <tt class="rst-docutils literal">self.failureException</tt>.</td>
  </tr>
</table>
          

          
      </div>

      <div id="childList">

        <div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.noCurrentResult">
    
  </a>
  <a name="noCurrentResult">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">async def</span> <span class="py-defname">noCurrentResult</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1202">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div><p class="undocumented">Undocumented</p></div>
  </div>
</div><div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.raisesException">
    
  </a>
  <a name="raisesException">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">async def</span> <span class="py-defname">raisesException</span>(self, exception=<a href="https://docs.python.org/3/library/constants.html#None" class="intersphinx-link">None</a>):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1205">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div><p class="undocumented">Undocumented</p></div>
  </div>
</div><div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.successResult">
    
  </a>
  <a name="successResult">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">async def</span> <span class="py-defname">successResult</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1199">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div><p class="undocumented">Undocumented</p></div>
  </div>
</div><div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_assertNoResult">
    
  </a>
  <a name="test_assertNoResult">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">test_assertNoResult</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1405">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div>When passed a coroutine with no current result, <code><a href="twisted.trial._synctest._Assertions.html#assertNoResult" class="internal-link" title="twisted.trial._synctest._Assertions.assertNoResult">SynchronousTestCase.assertNoResult</a></code> does not raise an exception.</div>
  </div>
</div><div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_assertNoResultFailure">
    
  </a>
  <a name="test_assertNoResultFailure">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">test_assertNoResultFailure</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1395">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div>When passed a coroutine which currently has an exception result (see <code>test_withFailureResult</code>), <code><a href="twisted.trial._synctest._Assertions.html#assertNoResult" class="internal-link" title="twisted.trial._synctest._Assertions.assertNoResult">SynchronousTestCase.assertNoResult</a></code> raises <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code>.</div>
  </div>
</div><div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_assertNoResultPropagatesLaterFailure">
    
  </a>
  <a name="test_assertNoResultPropagatesLaterFailure">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">test_assertNoResultPropagatesLaterFailure</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1429">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div>When passed a coroutine awaiting a <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.internet.defer.Deferred.html" class="intersphinx-link">Deferred</a></code> with no current result, which is then fired with a <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.python.failure.Failure.html" class="intersphinx-link">Failure</a></code> result, <code><a href="twisted.trial._synctest._Assertions.html#assertNoResult" class="internal-link" title="twisted.trial._synctest._Assertions.assertNoResult">SynchronousTestCase.assertNoResult</a></code> doesn't modify the result of the <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.internet.defer.Deferred.html" class="intersphinx-link">Deferred</a></code>.</div>
  </div>
</div><div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_assertNoResultPropagatesSuccess">
    
  </a>
  <a name="test_assertNoResultPropagatesSuccess">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">test_assertNoResultPropagatesSuccess</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1412">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div>When passed a coroutine awaiting a <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.internet.defer.Deferred.html" class="intersphinx-link">Deferred</a></code> with no current result, which is then fired with a success result, <code><a href="twisted.trial._synctest._Assertions.html#assertNoResult" class="internal-link" title="twisted.trial._synctest._Assertions.assertNoResult">SynchronousTestCase.assertNoResult</a></code> doesn't modify the result of the <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.internet.defer.Deferred.html" class="intersphinx-link">Deferred</a></code>.</div>
  </div>
</div><div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_assertNoResultSuccess">
    
  </a>
  <a name="test_assertNoResultSuccess">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">test_assertNoResultSuccess</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1385">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div>When passed a coroutine which currently has a success result (see <code>test_withSuccessResult</code>), <code><a href="twisted.trial._synctest._Assertions.html#assertNoResult" class="internal-link" title="twisted.trial._synctest._Assertions.assertNoResult">SynchronousTestCase.assertNoResult</a></code> raises <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code>.</div>
  </div>
</div><div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_assertNoResultSwallowsImmediateFailure">
    
  </a>
  <a name="test_assertNoResultSwallowsImmediateFailure">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">test_assertNoResultSwallowsImmediateFailure</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1447">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div>When passed a <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.internet.defer.Deferred.html" class="intersphinx-link">Deferred</a></code> which currently has a <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.python.failure.Failure.html" class="intersphinx-link">Failure</a></code> result, <code><a href="twisted.trial._synctest._Assertions.html#assertNoResult" class="internal-link" title="twisted.trial._synctest._Assertions.assertNoResult">SynchronousTestCase.assertNoResult</a></code> changes the result of the <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.internet.defer.Deferred.html" class="intersphinx-link">Deferred</a></code> to a success.</div>
  </div>
</div><div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_failureResultOfWithException">
    
  </a>
  <a name="test_failureResultOfWithException">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">test_failureResultOfWithException</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1373">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div>When passed a coroutine which currently has an exception result (ie, if converted into a <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.internet.defer.Deferred.html" class="intersphinx-link">Deferred</a></code>, <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.internet.defer.Deferred.html#addErrback" class="intersphinx-link">Deferred.addErrback</a></code> would cause the added errback to be called before <tt class="rst-docutils literal">addErrback</tt> returns), <code><a href="twisted.trial._synctest._Assertions.html#failureResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.failureResultOf">SynchronousTestCase.failureResultOf</a></code> returns returns a <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.python.failure.Failure.html" class="intersphinx-link">Failure</a></code> containing that exception.</div>
  </div>
</div><div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_failureResultOfWithExpectedException">
    
  </a>
  <a name="test_failureResultOfWithExpectedException">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">test_failureResultOfWithExpectedException</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1358">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div>When passed a coroutine which currently has an exception result (ie, if converted into a <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.internet.defer.Deferred.html" class="intersphinx-link">Deferred</a></code>, <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.internet.defer.Deferred.html#addErrback" class="intersphinx-link">Deferred.addErrback</a></code> would cause the added errback to be called before <tt class="rst-docutils literal">addErrback</tt> returns), <code><a href="twisted.trial._synctest._Assertions.html#failureResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.failureResultOf">SynchronousTestCase.failureResultOf</a></code> returns a <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.python.failure.Failure.html" class="intersphinx-link">Failure</a></code> containing that exception, if the exception type is expected.</div>
  </div>
</div><div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_failureResultOfWithoutResult">
    
  </a>
  <a name="test_failureResultOfWithoutResult">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">test_failureResultOfWithoutResult</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1243">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div><code><a href="twisted.trial._synctest._Assertions.html#failureResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.failureResultOf">SynchronousTestCase.failureResultOf</a></code> raises <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code> when called with a coroutine with no current result.</div>
  </div>
</div><div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_failureResultOfWithSuccess">
    
  </a>
  <a name="test_failureResultOfWithSuccess">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">test_failureResultOfWithSuccess</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1253">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div><code><a href="twisted.trial._synctest._Assertions.html#failureResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.failureResultOf">SynchronousTestCase.failureResultOf</a></code> raises <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code> when called with a coroutine with a success result.</div>
  </div>
</div><div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_failureResultOfWithWrongExceptionMultiExpectedExceptions">
    
  </a>
  <a name="test_failureResultOfWithWrongExceptionMultiExpectedExceptions">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">test_failureResultOfWithWrongExceptionMultiExpectedExceptions</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1309">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div><code><a href="twisted.trial._synctest._Assertions.html#failureResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.failureResultOf">SynchronousTestCase.failureResultOf</a></code> raises <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code> when called with a coroutine that raises an exception of a type that was not expected, and the <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code> message contains expected exception types in the error message.</div>
  </div>
</div><div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_failureResultOfWithWrongExceptionMultiExpectedExceptionsHasTB">
    
  </a>
  <a name="test_failureResultOfWithWrongExceptionMultiExpectedExceptionsHasTB">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">test_failureResultOfWithWrongExceptionMultiExpectedExceptionsHasTB</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1330">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div><code><a href="twisted.trial._synctest._Assertions.html#failureResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.failureResultOf">SynchronousTestCase.failureResultOf</a></code> raises <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code> when called with a coroutine that raises an exception of a type that was not expected, and the <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code> message contains the original exception traceback in the error message.</div>
  </div>
</div><div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_failureResultOfWithWrongExceptionOneExpectedException">
    
  </a>
  <a name="test_failureResultOfWithWrongExceptionOneExpectedException">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">test_failureResultOfWithWrongExceptionOneExpectedException</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1276">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div><code><a href="twisted.trial._synctest._Assertions.html#failureResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.failureResultOf">SynchronousTestCase.failureResultOf</a></code> raises <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code> when called with a coroutine that raises an exception with a failure type that was not expected, and the <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code> message contains the expected exception type.</div>
  </div>
</div><div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_failureResultOfWithWrongExceptionOneExpectedExceptionHasTB">
    
  </a>
  <a name="test_failureResultOfWithWrongExceptionOneExpectedExceptionHasTB">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">test_failureResultOfWithWrongExceptionOneExpectedExceptionHasTB</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1294">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div><code><a href="twisted.trial._synctest._Assertions.html#failureResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.failureResultOf">SynchronousTestCase.failureResultOf</a></code> raises <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code> when called with a coroutine that raises an exception with a failure type that was not expected, and the <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code> message contains the original exception traceback.</div>
  </div>
</div><div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_failureResultOfWithWrongFailure">
    
  </a>
  <a name="test_failureResultOfWithWrongFailure">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">test_failureResultOfWithWrongFailure</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1263">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div><code><a href="twisted.trial._synctest._Assertions.html#failureResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.failureResultOf">SynchronousTestCase.failureResultOf</a></code> raises <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code> when called with a coroutine that raises an exception that was not expected.</div>
  </div>
</div><div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_successResultOfWithException">
    
  </a>
  <a name="test_successResultOfWithException">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">test_successResultOfWithException</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1220">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div><code><a href="twisted.trial._synctest._Assertions.html#successResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.successResultOf">SynchronousTestCase.successResultOf</a></code> raises <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code> when called with a coroutine that raises an exception.</div>
  </div>
</div><div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_successResultOfWithFailureHasTraceback">
    
  </a>
  <a name="test_successResultOfWithFailureHasTraceback">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">test_successResultOfWithFailureHasTraceback</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1230">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div><code><a href="twisted.trial._synctest._Assertions.html#successResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.successResultOf">SynchronousTestCase.successResultOf</a></code> raises a <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code> that has the original failure traceback when called with a coroutine with a failure result.</div>
  </div>
</div><div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_successResultOfWithSuccessResult">
    
  </a>
  <a name="test_successResultOfWithSuccessResult">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">test_successResultOfWithSuccessResult</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1349">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div>When passed a coroutine which currently has a result (ie, if converted into a <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.internet.defer.Deferred.html" class="intersphinx-link">Deferred</a></code>, <code><a href="https://pydocbrowser.github.io/twisted/latest/twisted.internet.defer.Deferred.html#addCallback" class="intersphinx-link">Deferred.addCallback</a></code> would cause the added callback to be called before <tt class="rst-docutils literal">addCallback</tt> returns), <code><a href="twisted.trial._synctest._Assertions.html#successResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.successResultOf">SynchronousTestCase.successResultOf</a></code> returns that result.</div>
  </div>
</div><div class="basemethod">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.test_withoutResult">
    
  </a>
  <a name="test_withoutResult">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-keyword">def</span> <span class="py-defname">test_withoutResult</span>(self):
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1210">
      
      (source)
    </a>
  </div>
  <div class="docstring functionBody">
    
    
    <div><code><a href="twisted.trial._synctest._Assertions.html#successResultOf" class="internal-link" title="twisted.trial._synctest._Assertions.successResultOf">SynchronousTestCase.successResultOf</a></code> raises <code><a href="twisted.trial.unittest.SynchronousTestCase.html#failureException" class="internal-link" title="twisted.trial.unittest.SynchronousTestCase.failureException">SynchronousTestCase.failureException</a></code> when called with a coroutine with no current result.</div>
  </div>
</div><div class="baseclassvariable">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.exception">
    
  </a>
  <a name="exception">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-defname">exception</span> =
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1196">
      
      (source)
    </a>
  </div>
  <div class="functionBody">
    
    
    <div><p class="undocumented">Undocumented</p></div>
    
  </div>
</div><div class="baseclassvariable">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.failure">
    
  </a>
  <a name="failure">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-defname">failure</span> =
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1197">
      
      (source)
    </a>
  </div>
  <div class="functionBody">
    
    
    <div><p class="undocumented">Undocumented</p></div>
    
  </div>
</div><div class="baseclassvariable">
  
  
  <a name="twisted.trial.test.test_assertions.ResultOfCoroutineAssertionsTests.result">
    
  </a>
  <a name="result">
    
  </a>
  <div class="functionHeader">
    
    <span class="py-defname">result</span> =
    <a class="sourceLink" href="https://github.com/pydocbrowser/pydocbrowser.github.io/tree/main/build/sources/twisted-22.1.0//src/twisted/trial/test/test_assertions.py#L1195">
      
      (source)
    </a>
  </div>
  <div class="functionBody">
    
    
    <div><p class="undocumented">Undocumented</p></div>
    
  </div>
</div>

      </div>
    </div>

    <footer class="navbar navbar-default">
  
  <div class="container">
    <a href="index.html">API Documentation</a> for twisted,
  generated by <a href="https://github.com/twisted/pydoctor/">pydoctor</a>
    22.2.2.dev0 at 2022-02-28 16:15:31.
  </div>
</footer>

    <script src="pydoctor.js" type="text/javascript"></script>

  </body>
</html>